src/engine/MaxConstraint.cpp

```diff
@@ -121,64 +123,83 @@ void MaxConstraint::registerAsWatcher( ITableau *tableau )
 
-    if ( existsLowerBound( variable ) && !FloatUtils::gt( value, getLowerBound( variable ) ) )
+    if ( existsLowerBound( variable ) &&
+         !FloatUtils::gt( value, getLowerBound( variable ) ) )
         return;
 
     setLowerBound( variable, value );
 
-    bool maxErased = false;
-
-    if ( _elements.exists( variable ) && FloatUtils::gt( value, _maxLowerBound ) )
+    /*
+      See if we can eliminate any cases.
+    */
+    if ( _auxToElement.exists( variable ) )
     {
-        _maxLowerBound = value;
-        List<unsigned> toRemove;
-        for ( auto element : _elements )
+        // The case that this variable is an aux variable.
+        // We can eliminate the corresponding case if the aux variable is
+        // positive.
+        if ( FloatUtils::isPositive( value ) )
+            eliminateCase( _auxToElement[variable] );
+    }
+    else if ( variable == _f || _elements.exists( variable ) )
+    {
+        // If the variable is either in _element or _f. The only case that this is
+        // going to eliminate case is that the new lowerBound is greater than the
+        // _maxLowerBound.
+        // We update _maxLowerBound, and go through each element to see if it
+        // can be eliminated.
+        // We also update the _haveFeasibleEliminatedCase flag.
+        if ( FloatUtils::gt( value, _maxLowerBound ) )
         {
-            if ( element == variable || element == _f )
-                continue;
-            if ( existsUpperBound( element ) &&
-                 FloatUtils::lt( getUpperBound( element ), value ) )
+            _maxLowerBound = value;
+            List<unsigned> toRemove;
+            for ( auto element : _elements )
             {
-                toRemove.append( element );
+                if ( element == variable )
+                    continue;
+                if ( existsUpperBound( element ) &&
+                     FloatUtils::lt( getUpperBound( element ), value ) )
+                {
+                    toRemove.append( element );
+                }
             }
-        }
-
-        for ( unsigned removeVar : toRemove )
-        {
-            if ( _cdInfeasibleCases )
-                markInfeasible( variableToPhase( variable ) );
-            else
-                _elements.erase( removeVar );
+            for ( unsigned removeVar : toRemove )
+                eliminateCase( removeVar );
 
-            if ( maxIndexSet() && getMaxIndex() == removeVar )
-                maxErased = true;
+            _haveFeasibleEliminatedPhases =
+                FloatUtils::lte( _maxLowerBound,
+                                 _maxValueOfEliminatedPhases );
         }
     }
 
-    if ( maxErased )
-        resetMaxIndex();
+    if ( phaseFixed() )
+        _phaseStatus = ( _haveFeasibleEliminatedPhases ?
+                         MAX_PHASE_ELIMINATED :
+                         variableToPhase( *_elements.begin() ) );
 
+    /*
+      Now do some bound tightening.
+    */
```

Comment:
Re consistency: To clarify my previous comment, I think this is an OK use of the /* */ comments.